### PR TITLE
[react] RE: Add unstable_Profiler

### PR DESCRIPTION
**Pull request by eps1lon**
Copy of #33996

Any issue I can track about the state of this repo? Most of my PRs are now handled by a bot only and humans can only give advise that's essentially ignored. I do not understand how it is acceptable that the bot can close a PR that is accepted by an owner.

### CODE REVIEW

**Review by ferdaber — APPROVED** [review]

**Comment by ferdaber** [comment]
@eps1lon if local tests pass just cc @sandersn so he can merge manually.

**Comment by eps1lon** [comment]
/cc @sandersn @Jessidhia

CI failure is unrelated. Could any human stop the bot please?

**Comment by cscleison** [comment]
@eps1lon is there any workaround for typing `Set`? I need to target `es5`. I know I can just do `"lib": "es6"` but that implies the es6 apis are available at runtime (for ie11 I would have to add polyfills myself because `lib` it's just to tell typescript to not complain).

how to reproduce:
```
{
  "name": "test",
  "devDependencies": {
    "@types/react": "^16.9.1",
    "typescript": "^3.5.3"
  },
  "dependencies": {
    "react": "^16.9.0"
  }
}
```
```
{
  "compilerOptions": {
    "target": "es5",
    "strict": true,
    "jsx": "react",
    "outDir": "dist"
  },
  "include": ["./*.tsx"]
}
```
```

import * as React from "react";

const Foo = () => <div>bar</div>;

export { Foo };
```

then run `npx tsc`

**Comment by cscleison** [comment]
my workaround in the end was adding this, which i think is acceptable because `Map` and `Set` is partially supported by IE

`"lib": ["es5", "dom", "es2015.promise", "es2015.collection"]`

**Comment by eps1lon** [comment]
@cscleison Be aware that [you always have to polyfill `Map` and `Set` when using react](https://reactjs.org/docs/javascript-environment-requirements.html). IE11 does not support all methods properly so you still need add polyfills for those missing methods.

File: types/react/index.d.ts [hunk 1]
```diff
@@ -356,6 +366,26 @@ declare namespace React {
     const Suspense: ExoticComponent<SuspenseProps>;
     const version: string;
 
+    /**
+     * {@link https://github.com/bvaughn/rfcs/blob/profiler/text/0000-profiler.md#detailed-design | API}
```
Comment (intgr) [inline types/react/index.d.ts:370]: FYI this is a broken link.

Comment (intgr) [reply types/react/index.d.ts:370]: Opened a new PR #52735

**Pull request merged**